### ISSUE TITLE
Wrong icon preview shown for CupertinoIcons.location

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#3997]
See https://github.com/flutter/flutter/issues/104973:

![Screenshot 2022-05-31 at 10 09 26](https://user-images.githubusercontent.com/1078012/171137554-1bf6dbaa-5802-4c4a-8bdd-29eab3497552.png)

**Comment by DanTup** [comment]
Looks like this was a previous bug in Flutter that was fixed in https://github.com/flutter/flutter/pull/101923, but the icons in tools_metadata haven't been updated since.

The tools_metadata icons will need regenerating and shipping in VS Code.

@stevemessick FYI - not sure if this affects IntelliJ. I'll send a PR to tools_metadata shortly.

**Comment by stevemessick** [comment]
Thanks @DanTup I'm planning to update colors and icons in the upcoming release.

For background, the IntelliJ plugin does read the icons from the *.ttf or *.otf file. However, the Flutter 3.0 files have a lot of problems, so I'm going to revert to using the previews from tools_metadata for version 3.0 and up. At least until I get some time to look into what the cause of those problems is. It might be I have a bug to fix, but I suspect the Java code that reads the files isn't working.

**Comment by stevemessick** [comment]
I think the big stumbling block is reading the font file to create glyphs for icons. Java and Flutter have built-in support for that, but Dart does not. Flutter uses native code in its Sky engine. We'd either have to add support to Dart to read *.otf and *.ttf files (Material uses OTF and Cupertino uses TTF), or implement a Dart package to read font files and create glyphs. I have no idea how hard that would be, but given that everyone else seems to use C++ to do it, I don't expect it to be easy.

However, if we had *that* then it would be fairly straight-forward. The IntelliJ plugin has code to create icon previews that could be used to help design the analyzer version. Of course, it works with the IntelliJ PSI tree and the analyzer has its own version of syntax tree, but the concepts are similar (IIRC, it has been awhile since I looked at the analyzer).

**Comment by DanTup** [comment]
Yeah, there was some discussion about having `flutter` do this. Although it's not clear _when_ it would do it. For built-in icons it could be done as part of Flutter's initial initialization, but but it would make that slower. If we supported them in packages (which I suspect we'd like to), we'd need some way to trigger this on `flutter pub get` (but again, it could slow that down).

I'll think a bit more about this and may try to revive some discussion on https://github.com/dart-lang/sdk/issues/47667. Thanks!
